Link error with shared library and memory_resource inline members

When building Boost.JSON with `link=shared` (b2), this link error appears:
```
msvc.link C:\Users\vinnie\src\boost\bin.v2\libs\json\bench\msvc-14.2\debug\threading-multi\bench.exe
bench.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: void __thiscall boost::container::pmr::memory_resource::deallocate(void *,unsigned int,unsigned int)" (__imp_?deallocate@memory_resource@pmr@container@boost@@QAEXPAXII@Z) referenced in function "public: void __thiscall boost::json::detail::string_impl::destroy(class boost::json::storage_ptr const &)" (?destroy@string_impl@detail@json@boost@@QAEXABVstorage_ptr@34@@Z)
C:\Users\vinnie\src\boost\bin.v2\libs\json\bench\msvc-14.2\debug\threading-multi\bench.exe : fatal error LNK1120: 1 unresolved externals
```

We believe it is because the inline member function definitions are not being generated and exported in the resulting shared library. A possible solution is to decorate the declaration with the export macro.

--- a/include/boost/container/pmr/memory_resource.hpp
+++ b/include/boost/container/pmr/memory_resource.hpp
@@ -27,7 +27,7 @@ namespace pmr {
 
 //! The memory_resource class is an abstract interface to an
 //! unbounded set of classes encapsulating memory resources.
-class BOOST_CONTAINER_DECL memory_resource
+class memory_resource
 {
    public:
    // For exposition only

